tests/imfile-file-descriptor-exhasution.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+# This file is part of the rsyslog testbench, released  under ASL 2.0
+. ${srcdir:=.}/diag.sh init
+. $srcdir/diag.sh check-inotify
+
+generate_conf
+add_conf '
+# comment out if you need more debug info:
+	global( debug.whitelist="on"
+		debug.files=["imfile.c"])
+
+module(load="../plugins/imfile/.libs/imfile"
+       mode="inotify" normalizePath="off")
+
+input(type="imfile" File="./'$RSYSLOG_DYNNAME'.input.*.log" Tag="file:"
+	Severity="info" Facility="local7")
+'
+# generate input files first. Note that rsyslog processes it as
+# soon as it start up (so the file should exist at that point).
+# creating 1024 files to be parsed by imfile
+for  (( i= 0 ; i < 100 ; ++i ))
+do
+	echo test >> $RSYSLOG_DYNNAME.input.$i.file;
+	ln -s $RSYSLOG_DYNNAME.input.$i.file $RSYSLOG_DYNNAME.input.$i.log
+done
+
+# Start rsyslog now
+startup
+
+#set fd limit of rsyslog to 1024
+pid=$(getpid)
+prlimit --pid $pid --nofile=1024:1024
+
+# keep adding files and check wether the FD exhausted
+
+num_of_files=0
+for (( i = 100; i < 1200 ; ++i ))
+do
+	echo test >> $RSYSLOG_DYNNAME.input.$i.file;
+	ln -s $RSYSLOG_DYNNAME.input.$i.file $RSYSLOG_DYNNAME.input.$i.log
+	if  content_check --check-only "failed to processed incoming connection" $RSYSLOG_DYNNAME.started || \
+		content_check --check-only "Too many open files" $RSYSLOG_DYNNAME.started; then
```

Comment:
shouldn't we do line 42 after shutdown as a mandatory check?

Reply:
We can keep the check here as i don't want to keep creating files if FD is already exhausted.

In addition we can add one check at the end for confirming if we have hit this condition, before exiting.

Let me know if you had the same thoughts.